### ISSUE TITLE
test_glfw_canvas_del Windows fatal exception: access violation

### ISSUE THREAD

**Issue by Korijn** [pygfx/wgpu-py#446]
I just ran `pytest tests` and got the following segmentation fault:

```sh
tests\test_gui_glfw.py ..Windows fatal exception: access violation

Current thread 0x00005364 (most recent call first):
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\glfw\__init__.py", line 1274 in destroy_window
  File "C:\Users\kvang\dev\wgpu-py\wgpu\gui\glfw.py", line 188 in _on_close
  File "C:\Users\kvang\dev\wgpu-py\wgpu\gui\glfw.py", line 303 in close
  File "C:\Users\kvang\dev\wgpu-py\wgpu\gui\base.py", line 161 in __del__
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 1875 in _run_once
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 596 in run_forever
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\asyncio\windows_events.py", line 316 in run_forever
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 629 in run_until_complete
  File "C:\Users\kvang\dev\wgpu-py\tests\test_gui_glfw.py", line 83 in test_glfw_canvas_del
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\python.py", line 194 in pytest_pyfunc_call
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_callers.py", line 39 in _multicall
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_manager.py", line 80 in _hookexec
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_hooks.py", line 265 in __call__
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\python.py", line 1792 in runtest
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 169 in pytest_runtest_call
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_callers.py", line 39 in _multicall
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_manager.py", line 80 in _hookexec
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_hooks.py", line 265 in __call__
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 262 in <lambda>
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 341 in from_call
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 261 in call_runtest_hook
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 222 in call_and_report
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 133 in runtestprotocol
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\runner.py", line 114 in pytest_runtest_protocol
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_callers.py", line 39 in _multicall
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_manager.py", line 80 in _hookexec
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_hooks.py", line 265 in __call__
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\main.py", line 350 in pytest_runtestloop
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_callers.py", line 39 in _multicall
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_manager.py", line 80 in _hookexec
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_hooks.py", line 265 in __call__
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\main.py", line 325 in _main
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\main.py", line 271 in wrap_session
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\main.py", line 318 in pytest_cmdline_main
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_callers.py", line 39 in _multicall
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_manager.py", line 80 in _hookexec
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\pluggy\_hooks.py", line 265 in __call__
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\config\__init__.py", line 169 in main
  File "C:\Users\kvang\dev\wgpu-py\.venv\lib\site-packages\_pytest\config\__init__.py", line 192 in console_main
  File "C:\Users\kvang\dev\wgpu-py\.venv\Scripts\pytest.exe\__main__.py", line 7 in <module>
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87 in _run_code
  File "C:\Users\kvang\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197 in _run_module_as_main
...                                                                                  [ 28%]

...

=============================================== 120 passed in 8.55s ================================================
```

Interestingly the test does pass

**Comment by almarklein** [comment]
Is this reproducible?

**Comment by Korijn** [comment]
I'll try again tomorrow with the fault handler enabled.

**Comment by almarklein** [comment]
It now crashes at `is_closed`, where there is indeed an unguarded use of `_window`. I pushed a new commit to the `glfw` branch. Can you try again?

**Comment by Korijn** [comment]
With the last commit of the `glfw` branch it's fixed 🎉

```sh
❯ .venv/Scripts/pytest tests -k test_gui_glfw -s
================================================= test session starts =================================================
platform win32 -- Python 3.9.9, pytest-7.4.3, pluggy-1.0.0
rootdir: C:\Users\kvang\dev\wgpu-py
collecting ... _determine_can_use_wgpu_lib() status code: 0
collected 120 items / 115 deselected / 5 selected

tests\test_gui_glfw.py .....

========================================== 5 passed, 115 deselected in 4.49s ==========================================
```

**Comment by almarklein** [comment]
Closed by #448
